applications/main/clock/clock_app.c

```diff
@@ -0,0 +1,197 @@
+
+    if(state->settings.time_format == H24) {
+        snprintf(
+            time_string,
+            TIME_LEN,
+            CLOCK_TIME_FORMAT,
+            state->datetime.hour,
+            state->datetime.minute,
+            state->datetime.second);
+    } else {
+        bool pm = state->datetime.hour > 12;
+        snprintf(
+            time_string,
+            TIME_LEN,
+            CLOCK_TIME_FORMAT,
+            pm ? state->datetime.hour - 12 : state->datetime.hour,
+            state->datetime.minute,
+            state->datetime.second);
+
+        snprintf(
+            meridian_string,
+            MERIDIAN_LEN,
+            MERIDIAN_FORMAT,
+            pm ? MERIDIAN_STRING_PM : MERIDIAN_STRING_AM);
+    }
+
+    if(state->settings.date_format == Iso) {
+        snprintf(
+            date_string,
+            DATE_LEN,
+            CLOCK_ISO_DATE_FORMAT,
+            state->datetime.year,
+            state->datetime.month,
+            state->datetime.day);
+    } else {
+        snprintf(
+            date_string,
+            DATE_LEN,
+            CLOCK_RFC_DATE_FORMAT,
+            state->datetime.day,
+            state->datetime.month,
+            state->datetime.year);
+    }
+
+    furi_mutex_release(state->mutex);
+
+    canvas_set_font(canvas, FontBigNumbers);
+    canvas_draw_str_aligned(canvas, 64, 28, AlignCenter, AlignCenter, time_string);
+    canvas_set_font(canvas, FontSecondary);
+    canvas_draw_str_aligned(canvas, 64, 42, AlignCenter, AlignTop, date_string);
+
+    if(state->settings.time_format == H12)
+        canvas_draw_str_aligned(canvas, 65, 12, AlignCenter, AlignCenter, meridian_string);
+
+    FURI_LOG_T(TAG, "clock_render_callback - end");
+}
+
+static void clock_state_init(ClockState* const state) {
+    bool load_success = LOAD_CLOCK_SETTINGS(&state->settings);
+    if(!load_success) {
+        state->settings = DEFAULT_SETTINGS;
+    }
+    FURI_LOG_I(TAG, "Time format: %s", state->settings.time_format == H12 ? "12h" : "24h");
+    FURI_LOG_I(
+        TAG, "Date format: %s", state->settings.date_format == Iso ? "ISO 8601" : "RFC 5322");
+    furi_hal_rtc_get_datetime(&state->datetime);
+}
+
+// Runs every 1000ms by default
+static void clock_tick(void* ctx) {
+    furi_assert(ctx);
+    FuriMessageQueue* event_queue = ctx;
+    PluginEvent event = {.type = EventTypeTick};
+    // It's OK to loose this event if system overloaded
+    furi_message_queue_put(event_queue, &event, 0);
+}
+
+int32_t clock_app(void* p) {
+    UNUSED(p);
+    ClockState* plugin_state = malloc(sizeof(ClockState));
+
+    plugin_state->event_queue = furi_message_queue_alloc(8, sizeof(PluginEvent));
+    if(plugin_state->event_queue == NULL) {
+        FURI_LOG_E(TAG, "Cannot create event queue");
+        free(plugin_state);
+        return 255;
+    }
+    FURI_LOG_D(TAG, "Event queue created");
+
+    plugin_state->mutex = furi_mutex_alloc(FuriMutexTypeNormal);
+    if(plugin_state->mutex == NULL) {
```

Comment:
Never going to happen. 
All alloc functions are always return valid instance or crash system.